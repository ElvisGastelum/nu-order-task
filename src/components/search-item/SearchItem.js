import React from 'react';

import { key } from 'hooks/use-arrow-keys/key-code';

import {
  SearchItemLi,
  SearchItemLink
} from 'components/search-item/styled'

/**
 * Render item of suggestions
 *
 * @param {{ item: Object, className: string, index: number, setArrowKeys: (keyCode: number, index?: number) => void}} props used for the component
 * @return {JSX.Element} the JSX.Element
 */
export default function SearchItem({ item, active, setArrowKeys, index }) {

  const handleMouseEvent = e => setArrowKeys(key.MOUSE_EVENT, index);

  return (
    <SearchItemLi onMouseEnter={handleMouseEvent} active={active}>
      <SearchItemLink
        target="_blank"
        rel="noopener noreferrer"
        href={item.html_url}
      >
        {item.title}
      </SearchItemLink>
    </SearchItemLi>
  )
}
