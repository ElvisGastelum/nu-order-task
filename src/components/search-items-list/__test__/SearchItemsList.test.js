import React from 'react';

import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { act } from 'react-dom/test-utils';

import SearchItem from 'components/search-item'
import SearchItemList from 'components/search-items-list'
import { searchIssues } from 'hooks/use-suggestions/searchIssues';
import data from './mockData.json';

Enzyme.configure({
  adapter: new Adapter(),
})


describe('Test <SearchItemsList /> component', () => {
  test('should be have active prop the first suggestion item', async () => {

    var searchResults = [];
    const newSearchResults = value => {
      searchResults = searchIssues(value, data);
    }
    const setArrowKeys = e => { }

    newSearchResults('react')

    const wrapper = mount(
      <SearchItemList
        items={searchResults}
        index={0}
        setArrowKeys={setArrowKeys}
      />
    );

    const initialProp = wrapper.find(SearchItem).first().props().active;


    act(() => {
      wrapper.setProps({
        index: 0
      })
    })

    expect(wrapper.find(SearchItem).first().props().active).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  test('should be change active prop from first to the second suggestion item', async () => {

    var searchResults = [];
    const newSearchResults = value => {
      searchResults = searchIssues(value, data);
    }
    const setArrowKeys = e => { }

    newSearchResults('react')

    const wrapper = mount(
      <SearchItemList
        items={searchResults}
        index={0}
        setArrowKeys={setArrowKeys}
      />
    );

    const initialProp = wrapper.find(SearchItem).first().props().active;


    act(() => {
      wrapper.setProps({
        index: 1
      })
    })

    expect(wrapper.find(SearchItem).first().props().active).not.toBe(initialProp);
    expect(wrapper.find(SearchItem).at(1).props().active).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  test('should be have active prop the last suggestion item', async () => {

    var searchResults = [];
    const newSearchResults = value => {
      searchResults = searchIssues(value, data);
    }
    const setArrowKeys = e => { }

    newSearchResults('react')

    const wrapper = mount(
      <SearchItemList
        items={searchResults}
        index={0}
        setArrowKeys={setArrowKeys}
      />
    );

    act(() => {
      wrapper.setProps({
        index: 4
      })
    })

    expect(wrapper.find(SearchItem).last().props().active).not.toBe(false);
    expect(wrapper).toMatchSnapshot();
  });


})