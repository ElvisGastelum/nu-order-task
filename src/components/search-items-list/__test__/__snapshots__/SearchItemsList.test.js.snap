// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <SearchItemsList /> component should be change active prop from first to the second suggestion item 1`] = `ReactWrapper {}`;

exports[`Test <SearchItemsList /> component should be have active prop the first suggestion item 1`] = `ReactWrapper {}`;

exports[`Test <SearchItemsList /> component should be have active prop the last suggestion item 1`] = `ReactWrapper {}`;
